File: 0567-permutation-in-string/0567-permutation-in-string.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        if (s1.length() > s2.length()) {
            return false;
        }
        
        int n = s1.size();
        
        sort(s1.begin(), s1.end());
        
        for(int i = 0; i <= s2.size() - n; i++) {
            string t2 = s2.substr(i, n);
            sort(t2.begin(), t2.end());
            
            if(t2 == s1)
                return true;
        }
        
        return false;
    }
};

File: 0567-permutation-in-string/0567-permutation-in-string.py
class Solution(object):
    def checkInclusion(self, s1, s2):
        """
        :type s1: str
        :type s2: str
        :rtype: bool
        """
        ws=len(s1)
        s1=''.join(sorted(s1))
        for i in range(len(s2) - ws + 1):
            if i+ws <= len(s2):
                temp=s2[i:i+ws]
                temp=''.join(sorted(temp))
                if temp==s1:
                    return True
            else:
                break
        return False
        